pydm/tests/test_data_plugins_import.py

```diff
@@ -18,11 +22,14 @@ def test_plugin_directory_loading(qapp):
     with open(os.path.join(cur_dir, 'plugin_foo.py'), 'w+') as handle:
         handle.write(fake_file)
         handle.flush()
-    # Load plugins
-    load_plugins_from_path([cur_dir], 'foo.py')
-    assert 'tst1' in plugin_modules
-    assert 'tst2' in plugin_modules
-    os.remove(os.path.join(cur_dir, 'plugin_foo.py'))
+
+    try:
+        # Load plugins
+        load_plugins_from_path([cur_dir], 'foo.py')
+        assert 'tst1' in plugin_modules
+        assert 'tst2' in plugin_modules
+    finally:
+        os.remove(os.path.join(cur_dir, 'plugin_foo.py'))
```

Comment:
Much better than before.
Is this a good place to use the tempfile context manager that Robert recently used in happi?

Reply:
It definitely would have been a good spot for it - agreed